### ISSUE TITLE
cptui_create_custom_post_types function not found

### ISSUE THREAD

**Issue by JREubanks** [WebDevStudios/custom-post-type-ui#655]
I am getting the error 

`Warning: call_user_func_array() expects parameter 1 to be a valid callback, function 'cptui_create_custom_post_types' not found or invalid function name in C:\xampp\htdocs\where2bwho2c.dev\wp-includes\class-wp-hook.php on line 298`

I believe this is caused by the CPT UI plugin, but the site is disabled so I can't deactivate it. I tried just deleting the plugin subdirectory, but still got the same error. I want this error to go away so I can start from scratch with the plugin. How can I do this?

**Comment by tw2113** [comment]
Something odd is definitely going on because that's a function found in primary file for the plugin. However you say the plugin itself isn't deactivated, so it should be available. Should be around line 244 in the 1.5.5 version of the plugin.

Best idea I have is that somehow the file got corrupted or only partially loaded.

For what it's worth, deleting the plugin from the server isn't going to restart fresh with the post types/taxonomies registered. The following link will show you how to delete the settings, but it's not going to delete actual posts/taxonomy terms.

http://docs.pluginize.com/article/20-resetting-cptui-settings

Have you checked any error logs for more information about other errors that may be occuring, outside of the one posted initially?

**Comment by JREubanks** [comment]
Thanks for your reply.

 

There is definitely something odd about this. Since the function in question (cptui_create_custom_post_types) is in your plugin, where is the request coming from 

in class-wp-hook.php if the plugin has been removed from the plugins folder altogether? Is this coming from something in the wp-options table?

 

If I delete your entries in the wp-options table then reinstall and activate your plugin, then re-enter my custom taxonomies, can I then recover my custom taxonomy content by restoring backups of the tables involved (wp_terms, wp_term_taxonomy, and wp_term_relationships)?

 

I tried copying a new copy of the plugin into the plugins folder but got the same results. I do not want to delete anything from the database because I have lots of content using custom taxonomies. I have gotten away from custom post types altogether since they were not performant, and am now using custom database tables, so custom post types are not a problem.

 

Thanks for your help.

 



Russell Eubanks

Rubin-Eubanks Associates

6495 S. Clarkson Street

Centennial, CO 80121-2417

720-219-3738

**Comment by tw2113** [comment]
Unless you delete the taxonomy terms/post type posts manually, they're still going to be present in the database without having to restore a backup. We consciously leave content in tact and in place, even if someone abandons the plugin completely. They should be able to re-register types/taxonomies on their own with a matching slug and be back to where they were before.

The only way I can think of for that error cropping up is if somehow an add_action call is being run before the plugin is even loaded. Do you have anything in an mu-plugins folder that may be getting loaded? Perhaps somewhere a line added that tries to run the registration process really early?

**Comment by JREubanks** [comment]
Yes, I found it. Previously, I had contacted you with a problem about custom post types not showing up in the admin menu. You had suggested I add

 

remove_action( 'init', 'cptui_create_custom_post_types', 10 );

add_action( 'init', 'cptui_create_custom_post_types', 11 );

 

This is what has been giving me fits. Thanks for your help.

 

 



Russell Eubanks

Rubin-Eubanks Associates

6495 S. Clarkson Street

Centennial, CO 80121-2417

720-219-3738
